sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -39,8 +39,29 @@ import org.apache.spark.util.{MutableURLClassLoader, Utils}
 
 /**
  * A class that holds all state shared across sessions in a given [[SQLContext]].
+ *
+ * @param sparkContext The Spark context associated with this SharedState
+ * @param initialConfigs The configs from the very first created SparkSession
  */
-private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
+private[sql] class SharedState(
+    val sparkContext: SparkContext,
+    initialConfigs: scala.collection.Map[String, String])
+  extends Logging {
+
+  private lazy val (conf, hadoopConf) = {
```

Comment:
Oh, I think Wenchen said https://github.com/apache/spark/pull/23709#discussion_r252921293 should not better be reflected. I am okay with his comment and it is okay without this change - sorry for forth and back.